### ISSUE TITLE
Enh: logger.Close() should return error

### ISSUE THREAD

**Issue by githubjeka** [go-ozzo/ozzo-log#15]
P.S. 
And some thoughts about 
```
logger.Open()
defer logger.Close()
```
https://joeshaw.org/dont-defer-close-on-writable-files/

**Comment by kolkov** [comment]
Good point, and thanks for the linked article — the general principle of checking errors on `Close()` for writable resources absolutely applies here. Losing the last batch of buffered log entries silently is not great.

I'm actively maintaining these packages now, and I agree `Close()` should return an error. Will address this alongside some other error-handling improvements in the logger.
